File: server/internal/domain/expense.go
package domain

import (
	"time"
)

// Expense represents a spending transaction against a budget envelope
type Expense struct {
	ID          int64     `json:"id"`
	BudgetID    int64     `json:"budget_id"`
	Amount      float64   `json:"amount"`
	Description string    `json:"description"`
	Date        time.Time `json:"date"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateExpenseRequest struct {
	BudgetID    int64   `json:"budget_id"`
	Amount      float64 `json:"amount"`
	Description string  `json:"description"`
	Date        string  `json:"date"` // Format: "2006-01-02"
}

type UpdateExpenseRequest struct {
	BudgetID    *int64   `json:"budget_id,omitempty"`
	Amount      *float64 `json:"amount,omitempty"`
	Description *string  `json:"description,omitempty"`
	Date        *string  `json:"date,omitempty"`
}

// ExpenseFilter for querying expenses
type ExpenseFilter struct {
	BudgetID  *int64
	StartDate *time.Time
	EndDate   *time.Time
}

File: iOS/Sources/Domain/Entities/Expense.swift
import Foundation

/// Expense represents a spending transaction against a budget envelope
struct Expense: Identifiable, Codable {
    let id: Int64
    var budgetId: Int64
    var amount: Double
    var description: String
    var date: Date
    var createdAt: Date
    var updatedAt: Date

    init(
        id: Int64 = 0,
        budgetId: Int64,
        amount: Double,
        description: String,
        date: Date = Date(),
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.budgetId = budgetId
        self.amount = amount
        self.description = description
        self.date = date
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
